### PR TITLE
feat / update Gateway endpoints & parameter formats

### PR DESCRIPTION
**Pull request by sdgoh**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:

- Add protocols swap pool caching on start
- Refactor Gateway API endpoints
- Update request param data to use token symbol instead of contract address look up

### CODE REVIEW

File: hummingbot/strategy/amm_arb/amm_arb.py [hunk 1]
```diff
@@ -253,13 +253,18 @@ async def format_status(self) -> str:
             market, trading_pair, base_asset, quote_asset = market_info
             buy_price = await market.get_quote_price(trading_pair, True, self._order_amount)
             sell_price = await market.get_quote_price(trading_pair, False, self._order_amount)
-            mid_price = (buy_price + sell_price) / 2
+
+            # check for unavailable price data
+            buy_price = float(buy_price) if buy_price is not None else '-'
+            sell_price = float(sell_price) if sell_price is not None else '-'
+            mid_price = float((buy_price + sell_price) / 2) if '-' not in [buy_price, sell_price] else '-'
```
Comment (vic-en) [inline hummingbot/strategy/amm_arb/amm_arb.py:260]:
```suggestion
            buy_price = Decimal(str(buy_price)) if buy_price is not None else '-'
            sell_price = Decimal(str(sell_price)) if sell_price is not None else '-'
            mid_price = ((buy_price + sell_price) / 2) if '-' not in [buy_price, sell_price] else '-'
```

File: hummingbot/strategy/amm_arb/data_types.py [hunk 2]
```diff
@@ -79,7 +79,9 @@ def profit_pct(self, account_for_fee: bool = False, first_side_quote_eth_rate: D
 
         sell_gained_net = (sell.amount * sell.quote_price) - sell_fee_amount
         buy_spent_net = (buy.amount * buy.quote_price) + buy_fee_amount
-        return ((sell_gained_net - buy_spent_net) / buy_spent_net) if buy_spent_net != s_decimal_0 else s_decimal_0
+        profit_percentage = ((sell_gained_net - buy_spent_net) / buy_spent_net) if buy_spent_net != s_decimal_0 else s_decimal_0
+
+        return profit_percentage
```
Comment (vic-en) [inline hummingbot/strategy/amm_arb/data_types.py:84]:
```suggestion
        return ((sell_gained_net - buy_spent_net) / buy_spent_net) if buy_spent_net != s_decimal_0 else s_decimal_0
```

**Review by vic-en — CHANGES_REQUESTED** [review]
Aside this suggestions, both `initiate_pool_task` and `auto_approve_task` basically do the same thing in a slightly different way. We should agree on one and remove the other.
Aside that, the rest of the changes LGTM.

File: hummingbot/connector/connector/balancer/balancer_connector.py [hunk 3]
```diff
@@ -255,24 +295,28 @@ async def _create_order(self,
...
             hash = order_result.get("txHash")
+            gas_price = order_result.get("gasPrice")
+            gas_limit = order_result.get("gasLimit")
+            gas_cost = order_result.get("gasCost")
+            self.start_tracking_order(order_id, None, trading_pair, trade_type, price, amount, gas_price)
             tracked_order = self._in_flight_orders.get(order_id)
+
+            # update onchain balance
+            await self._update_balances()
+
```
[the remark this answers is not in the document]
Comment (sdgoh) [reply hummingbot/connector/connector/balancer/balancer_connector.py:315]: I'll remove this call to update the balance as the Gateway will be updating the on-chain balance.

**Review by PtrckM — APPROVED** [review]
Test and confirmed working by QA Team.

**Comment by PtrckM** [comment]
@sdgoh @vic-en 
- `ethereum_token_list_url` still configurable on client via `config ethereum_token_list_url` and on `conf_global.yml`, these values are already set on `.env` via source and `gateway.env` via docker from gateway side.
- xdai provider on `./create-gateway.sh` is not configurable.

**Comment by vic-en** [comment]
>     * ./create-gateway.s

- The ethereum token list on client is there to prevent the client from blocking on startup.
- I pushed a commit to make XDAI rpc provider configurable. Kindly confirm.

**Review by PtrckM — APPROVED** [review]
No more issues encountered during testing. Confirmed working by QA.

**Review by vic-en — APPROVED** [review]
LGTM

**Review by vic-en — APPROVED** [review]
LGTM

**Pull request merged**